core/contacts.go

```diff
@@ -25,21 +25,49 @@ type ContactInfo struct {
 	ThreadIds []string    `json:"thread_ids,omitempty"`
 }
 
-// ContactInfoQuery describes a contact search query
-type ContactInfoQuery struct {
+// ContactQuery describes a contact search query
+type ContactQuery struct {
 	Id       string
 	Address  string
 	Username string
 	Local    bool
 	Limit    int
 	Wait     int
-	Add      bool
 }
 
-// ContactInfoQueryResult displays info about a contact search result
-type ContactInfoQueryResult struct {
-	Local  []ContactInfo `json:"local,omitempty"`
-	Remote []ContactInfo `json:"remote,omitempty"`
+// ContactQueryResult displays info about a contact search result
+type ContactQueryResult struct {
+	Local   bool        `json:"local"`
+	Contact ContactInfo `json:"contact"`
+}
+
+// contactSet holds a unique set of contact search results
+type contactSet struct {
```

Comment:
Provides a map based collection of contacts that can be added to by concurrent go routines. Ignores additions from older results.